### DISCUSSION TITLE
Why can't I use TimeFunctionComponentControl in HeatTransferFromHeatFlux1Phase?

### DISCUSSION THREAD

**Discussion by Ethan-xj** [idaholab/moose#26736] [Q&A Modules: Thermal Hydraulics]
@joshuahansel @GiudGiud Hello. I found that I couldn't use `TimeFunctionComponentControl` in `HeatTransferFromHeatFlux1Phase`.
I set the input file like:
```
  [FLOWSC1_POWER]
    type = HeatTransferFromHeatFlux1Phase
    flow_channel = core_chan1
    q_wall = 0
    P_hf = ${fparse pi*rod_diameter/2.0}
  [../]
  [set_power1_value]
    type = TimeFunctionComponentControl
    component = FLOWSC1_POWER
    parameter = q_wall
    function = PowerTran
  [../]
```
And in terminal:
```
*** ERROR ***
The following error occurred in the object "set_power1_value", of type "TimeFunctionComponentControl".
The desired parameter 'component/FLOWSC1_POWER/Hw' was not located for the 'set_power1_value' object, it either does not exist or has not been declared as controllable.
```
Is there anything wrong with my setting?

Ethan

**Comment by joshuahansel** [comment]
That component/parameter is currently incompatible with `TimeFunctionComponentControl`. Usually we make component parameters like this to be `Real` parameters (which we can make controllable), but this one is a `FunctionName` parameter. So you can just use your function directly:
```
q_wall = PowerTran
```
Just thinking out loud, I think we should make these parameters into `FunctionName` parameters because you can always just supply a `ConstantFunction` whose `value` parameter is controllable.

**Reply by Ethan-xj** [reply]
I got it. Thank you!

**Reply by Ethan-xj** [reply]
Oh wait, the PowerTran is a `PiecewiseLinear` function which is time-dependent. If I use `q_wall = PowerTran` directly, how should I express a time-dependent function using `PiecewiseLinear`?

**Reply by Ethan-xj** [reply]
Oh I see, the x corresponds to time by default.
